flexget/plugins/input/imdb_watchlist.py

```diff
@@ -84,72 +85,144 @@ def on_task_input(self, task, config):
         log.verbose('Retrieving imdb list: %s', config['list'])
 
-        params = {'view': 'compact', 'start': 1}
+        headers = {'Accept-Language': config.get('force_language')}
+        params = {'view': 'detail', 'page': 1}
         if config['list'] in ['watchlist', 'ratings', 'checkins']:
             url = 'http://www.imdb.com/user/%s/%s' % (config['user_id'], config['list'])
+            if config['list'] == 'watchlist':
+                params = {'view': 'detail'}
         else:
             url = 'http://www.imdb.com/list/%s' % config['list']
+        if 'all' not in config['type']:
+            title_types = []
+            for title_type in config['type']:
+                title_types.append(TITLE_TYPE_MAP[title_type])
+            params['title_type'] = ','.join(title_types)
+            params['sort'] = 'list_order%2Casc'
+
+        if config['list'] == 'watchlist':
+            entries = self.parse_react_widget(task, config, url, params, headers)
+        else:
+            entries = self.parse_html_list(task, config, url, params, headers)
+        return entries
 
-        headers = {'Accept-Language': config.get('force_language')}
+    def fetch_page(self, task, url, params, headers):
         log.debug('Requesting: %s %s', url, headers)
-
         try:
             page = task.requests.get(url, params=params, headers=headers)
         except HTTPError as e:
             raise plugin.PluginError(e.args[0])
         if page.status_code != 200:
             raise plugin.PluginError('Unable to get imdb list. Either list is private or does not exist.')
+        return page
 
-        soup = get_soup(page.text)
+    def parse_react_widget(self, task, config, url, params, headers):
+        page = self.fetch_page(task, url, params, headers)
+        try:
+            re_result = re.search('IMDbReactInitialState.push\((.+?)\);\n', page.text)
+            json_text = re_result.group(1)
+            json_vars = json.loads(json_text)
+        except (TypeError, AttributeError, ValueError) as e:
+            raise plugin.PluginError(
+                'Unable to get imdb list from imdb react widget.' +
+                ' Either the list is empty or the imdb parser plugin is broken.' +
+                ' Original error: %s.' % e.args[0])
+        total_item_count = 0
+        if 'list' in json_vars and 'items' in json_vars['list']:
+            total_item_count = len(json_vars['list']['items'])
+        if not total_item_count:
+            log.verbose('No movies were found in imdb list: %s', config['list'])
+            return
+        imdb_ids = []
+        for item in json_vars['list']['items']:
+            if is_valid_imdb_title_id(item['const']):
+                imdb_ids.append(item['const'])
+        params = {'ids': ','.join(imdb_ids)}
+        url = 'http://www.imdb.com/title/data'
+        try:
+            page = task.requests.get(url, params=params)
```

Comment:
If you give it the param `raise_status=True` it will raise an exception when the error code is 4xx, 5xx etc. I think that's what you're going for?

And fyi, Requests raises a RequestException, so catch that instead of HTTPError.